### PR TITLE
Pull if

### LINKED ISSUE #3308: API: PullPolicy string values have "Pull" prefix, but no other string constants do that
PodPhase: Pending, Running, etc

Protocol: TCP, UDP

DNSPolicy: ClusterFirst, Default

AffinityType: ClientIP, None

Even RestartPolicy { Always, Never }

PullPolicy is the lone odd man out.

### PR DESCRIPTION
**Pull request by thockin**
Minor point, and I could be talked out of it, but i wanted to put it up before v1b3 locks in.

Fixes #3308

### CODE REVIEW

**Comment by bgrant0607** [comment]
You want to do this to all API constants, such as RestartPolicy?

/cc @smarterclayton

**Comment by smarterclayton** [comment]
I like it being shorter - I assume you'll do conversion across internal to external?

**Comment by thockin** [comment]
@bgrant0607 RestartPolicy is already shortened: the fields are named "Always" and "Never".

@smarterclayton yeah, I'll convert for the older APIs since people seem to be in favor.  Forgot to do that (and yet no test failed).

**Comment by bgrant0607** [comment]
Oh, it has already been done to all other constants. Ok, yes, I'm definitely in favor.

**Comment by thockin** [comment]
Now with conversions and serialization_test hook.  It's really obnoxious how many times I need to spell out what valid random values are.

**Comment by smarterclayton** [comment]
Yes.

**Comment by smarterclayton** [comment]
LGTM

File: pkg/api/v1beta1/conversion.go [hunk 1]
```diff
@@ -707,6 +707,38 @@ func init() {
...
+			case newer.PullAlways:
+				*out = PullAlways
+			case newer.PullNever:
+				*out = PullNever
+			case newer.PullIfNotPresent:
+				*out = PullIfNotPresent
+			case "":
+				*out = ""
+			default:
+				//FIXME: Should we be doing validation here?  How does this play into the UX?
```
Comment (smarterclayton) [inline pkg/api/v1beta1/conversion.go]: Technically we shouldn't, unless the only place the check can be done is here.  I think we might eventually want a revalidation on the api version, then validate on internal.

**Comment by bgrant0607** [comment]
LGTM, also

File: pkg/api/v1beta1/conversion.go (hunk 1 above)
Comment (thockin) [inline pkg/api/v1beta1/conversion.go]: The alternative is just to let the invalid value through and let validation catch it.

I'm going to do that, and leave a note here try to tackle it as part of fixing user-facing error messages and versioning selectors.  OK?

**Comment by thockin** [comment]
Now with invalid error pass-through.  If it goes green I can merge it since I am on-call.

**Comment by smarterclayton** [comment]
Yeah, that's fine

**Pull request merged**